### ISSUE TITLE
Error when reading public API

### ISSUE THREAD

**Issue by soerenjalas** [toto/homebridge-airrohr#2]
Hi,
First of all thanks for the great plug-in!
I'm trying to use the plugin via the public API, but homebridge keeps failing with the following error

```
 [AirRohr] Could not get sensor data: TypeError: this.dataCache.sensordatavalues is not iterable
```

```json
{
    "accessory": "airrohr",
    "name": "AirRohr",
    "json_data": "http://api.luftdaten.info/v1/sensor/4668/",
    "sensor_id": "4668",
    "update_interval_seconds": 60
}
```

I checked the response of the API and it seems to be fine

```json
[{"sampling_rate":null,"sensordatavalues":[{"value_type":"P1","value":"36.50","id":1404263069},{"value_type":"P2","value":"11.77","id":1404263070}],"timestamp":"2018-01-04 20:26:13","id":644020044,"sensor":{"sensor_type":{"id":14,"manufacturer":"Nova Fitness","name":"SDS011"},"pin":"1","id":4668},"location":{"longitude":"10.044","latitude":"53.568","id":2348,"country":"DE"}},{"sampling_rate":null,"sensordatavalues":[{"value_type":"P1","value":"33.57","id":1404275471},{"value_type":"P2","value":"10.70","id":1404275472}],"timestamp":"2018-01-04 20:28:40","id":644025931,"sensor":{"sensor_type":{"id":14,"manufacturer":"Nova Fitness","name":"SDS011"},"pin":"1","id":4668},"location":{"longitude":"10.044","latitude":"53.568","id":2348,"country":"DE"}}]
```

Do you have an idea what could be the problem?

**Comment by honkmaster** [comment]
+1

**Comment by gerritfries** [comment]
Same for me. A fix would be great!

**Comment by toto** [comment]
This will indeed be fixed.

**Comment by honkmaster** [comment]
Another sample URL: http://api.luftdaten.info/v1/sensor/3139/

**Comment by toto** [comment]
v1.1 (just published on npm) should fix this. Note the updated config for public API vs internal network.

**Comment by honkmaster** [comment]
I think it needs to be reopened....

`TypeError: this.dataCache.updateFromAPI is not a function`

**Comment by toto** [comment]
Oh damn. This project needs need better tests I guess…

**Comment by toto** [comment]
Fixed in 1.1.3
